src/Sinch/Conversation/Conversations/Conversations.cs

```diff
@@ -0,0 +1,233 @@
+        }
+
+        /// <inheritdoc />
+        public Task Delete(string conversationId, CancellationToken cancellationToken = default)
+        {
+            if (string.IsNullOrEmpty(conversationId))
+                throw new ArgumentNullException(nameof(conversationId), "Should have a value");
+
+            var uri = new Uri(_baseAddress,
+                $"v1/projects/{_projectId}/conversations/{conversationId}");
+            _logger?.LogDebug("Deleting a {conversationId} of {project}", conversationId, _projectId);
+            return _http.Send<Conversation>(uri, HttpMethod.Delete,
+                cancellationToken);
+        }
+
+        /// <inheritdoc />
+        public Task Stop(string conversationId, CancellationToken cancellationToken = default)
+        {
+            if (string.IsNullOrEmpty(conversationId))
+                throw new ArgumentNullException(nameof(conversationId), "Should have a value");
+
+            var uri = new Uri(_baseAddress,
+                $"v1/projects/{_projectId}/conversations/{conversationId}:stop");
+            _logger?.LogDebug("Stopping a {conversationId} of {project}", conversationId, _projectId);
+            return _http.Send<Conversation>(uri, HttpMethod.Post,
+                cancellationToken);
+        }
+
+        /// <inheritdoc />
+        public Task<Conversation> Update(Conversation conversation,
+            MetadataUpdateStrategy metadataUpdateStrategy = null,
+            CancellationToken cancellationToken = default)
+        {
+            if (conversation == null)
+                throw new ArgumentNullException(nameof(conversation), "Conversation shouldn't be null");
+
+            if (string.IsNullOrEmpty(conversation.Id))
+                throw new NullReferenceException($"{nameof(conversation.Id)} should have a value");
+
+            var builder = new UriBuilder(new Uri(_baseAddress,
+                $"v1/projects/{_projectId}/conversations/{conversation.Id}"));
+
+            var queryString = HttpUtility.ParseQueryString(string.Empty);
+            var propMask = conversation.GetPropertiesMask();
+            if (!string.IsNullOrEmpty(propMask)) queryString.Add("update_mask", propMask);
+
+            if (metadataUpdateStrategy is not null)
+                queryString.Add("metadata_update_strategy", metadataUpdateStrategy.Value);
+
+            builder.Query = queryString?.ToString()!; // it's okay to pass null.
+
+            _logger?.LogDebug("Updating a {conversationId} of {project}", conversation.Id, _projectId);
+            return _http.Send<Conversation, Conversation>(builder.Uri, HttpMethod.Patch, conversation,
+                cancellationToken);
+        }
+
+        /// <inheritdoc />
+        public Task InjectMessage(InjectMessageRequest injectMessageRequest,
+            CancellationToken cancellationToken = default)
+        {
+            if (string.IsNullOrEmpty(injectMessageRequest.ConversationId))
+                throw new NullReferenceException(
```

Comment:
Sometimes you are throwing a `NullReferenceException` and other times it's a `ArgumentNullException`

Reply:
True - it's intended, `ArgumentNullException` is obviously for argument of a function, null ref for is for a other props of an object in this case.

Reply:
So in this case, you may also want to check the argument is not null before checking its properties in order to avoid a NPE